File: pi.java
public class pi {

  public static double pi_nks(int limit) {
    double pi = 3.0;
    int s = 1;
    for(int i=2; i<=limit; i+=2) {
      pi += (double) s*4 /  (double)  (i * (i+1) * (i+2) );
      s = (s*(-1));
    }
    return pi;
  }

  public static double pi_gls(int limit) {
    double pi = 0.0;
    int s = 1;
    for(int i=1; i<=limit; i+=2) {
      pi += s * (double) 4/i;
      s = (s*(-1));
    }
    return pi;
  }

  final static int LIMIT = 100;
  public static void main(String[] args) {
    System.out.printf("NKS: %.13f%n", pi_nks(LIMIT));
    System.out.printf("GLS: %.13f%n", pi_gls(LIMIT));
  }
}

File: pi.jl
function pi_nks(limit::Core.Int64)::Core.Float64
    pi = 3.0
    s = 1
    for i in range(start=2, stop=limit, step=2)
        pi += (s * 4 / (i * (i + 1) * (i + 2)))
        s = s * (-1)
    end
    return pi
end

function pi_gls(limit::Core.Int64)::Core.Float64
    pi = 0.0
    s = 1
    for i in range(start=1, stop=limit, step=2)
        pi += (s * (4 / i))
        s = s * (-1)
    end
    return pi
end

const limit = 100
println("NKS: $(round.(pi_nks(limit), digits=13))")
println("GLS: $(round.(pi_gls(limit), digits=13))")
